.github/workflows/reusable_dockerfile_pipeline.yml

```
@@ -135,23 +150,42 @@ jobs:
           severity: "CRITICAL,HIGH"
 
   docker-build:
+    name: docker-build (${{ matrix.registry.name }}; ${{ matrix.registry.registry-url }}/${{ matrix.registry.registry-owner }}/${{ needs.prepare-env.outputs.output_image_name }}:${{ needs.prepare-env.outputs.output_short_sha }})
     runs-on: "ubuntu-latest"
     # wait until the jobs are finished.
     needs: ["prepare-env", "logic-check", "docker-security"]
     permissions:
       contents: write
       packages: write
-
+    strategy:
+      matrix:
+        registry:
+          - name: DockerHub
+            user-secret: DOCKERHUB_USERNAME
+            token-secret: DOCKERHUB_TOKEN
+            registry-url: docker.io
+            registry-owner: celestiaorg
+          - name: GHCR
+            user-secret: ${{ github.repository_owner }}
+            token-secret: GITHUB_TOKEN
+            registry-url: ghcr.io
+            registry-owner: ${{ needs.prepare-env.outputs.repo_owner }}
+          - name: ScaleWay
+            user-secret: SCALEWAY_USERNAME
+            token-secret: SCW_SECRET_KEY
+            registry-url: rg.fr-par.scw.cloud
+            registry-owner: celestiaorg
+      fail-fast: false
     steps:
       - name: Checkout
         uses: "actions/checkout@v4"
 
-      - name: Login to GHCR
+      - name: Login to ${{ matrix.registry.name }}
         uses: docker/login-action@v3
         with:
-          registry: ${{ env.REGISTRY }}
-          username: ${{ github.repository_owner }}
-          password: ${{ secrets.GITHUB_TOKEN }}
+          registry: ${{ matrix.registry.registry-url }}
+          username: ${{ matrix.registry.registry-url == 'ghcr.io' && matrix.registry.user-secret || secrets[matrix.registry.user-secret] }}
```

Doesn't this evaluate to true or false? I don't think this is going to return a username.

---

also should we use the env variable?
```suggestion
          username: ${{ matrix.registry.registry-url == env.GITHUB_REG && matrix.registry.user-secret || secrets[matrix.registry.user-secret] }}
```

---

also username isn't really a secret, would it be simplier if none of the user names were secrets?

---

You have a point that users are not secrets. 
However, I'd argue that from another angle as creating env vars for reusable workflows in github's repos is pure pain 
https://github.com/orgs/community/discussions/26671#discussioncomment-5063618

As secrets can be inherited, it's much easier to pass this like I did here and not worry about it afterwards
https://github.com/Bidon15/celestia-node/blob/main/.github/workflows/docker-build-publish.yml

---

> As secrets can be inherited, it's much easier to pass this like I did here and not worry about it afterwards

I was saying to not use env or secrets. just hardcode the usernames to make it easier.